playground/backend/internal/preparators/java_preparators.go

```diff
@@ -0,0 +1,125 @@
+	"os"
+	"regexp"
+	"strings"
+)
+
+const (
+	classWithPublicModifierPattern    = "public class "
+	classWithoutPublicModifierPattern = "class "
+	packagePattern                    = `package ([\w]+\.)+[\w]+;`
+	emptyStringPattern                = ""
+	newLinePattern                    = "\n"
+	pathSeparatorPattern              = "/"
+	tmpFileSuffix                     = "tmp"
+)
+
+// GetJavaPreparation return preparation methods that should be applied to Java code
+func GetJavaPreparation(filePath string) *[]Preparator {
+	publicClassModification := Preparator{
+		Prepare: replace,
+		Args:    []interface{}{filePath, classWithPublicModifierPattern, classWithoutPublicModifierPattern},
+	}
+	additionalPackage := Preparator{
+		Prepare: replace,
+		Args:    []interface{}{filePath, packagePattern, emptyStringPattern},
+	}
+	preparation := []Preparator{publicClassModification, additionalPackage}
+	return &preparation
+}
+
+// replace process file by filePath and replace all patterns to newPattern
+func replace(args ...interface{}) error {
+	filePath := args[0].(string)
+	pattern := args[1].(string)
+	newPattern := args[2].(string)
+
+	file, err := os.Open(filePath)
+	if err != nil {
+		log.Printf("Preparation: Error during open file: %s, err: %s\n", filePath, err.Error())
+		return err
+	}
+	defer file.Close()
+
+	tmp, err := createTempFile(filePath)
+	if err != nil {
+		log.Printf("Preparation: Error during create new temporary file, err: %s\n", err.Error())
+		return err
+	}
+	defer tmp.Close()
+
+	reg := regexp.MustCompile(pattern)
+	scanner := bufio.NewScanner(file)
+
+	// uses to indicate when need to add new line to tmp file
+	newLine := false
+	for scanner.Scan() {
+		err = addNewLine(newLine, tmp)
+		if err != nil {
+			log.Printf("Preparation: Error during write \"%s\" to tmp file, err: %s\n", newLinePattern, err.Error())
+			return err
+		}
+		line := scanner.Text()
+		matches := reg.FindAllString(line, -1)
+		for _, str := range matches {
+			line = strings.ReplaceAll(line, str, newPattern)
+		}
+		if _, err = io.WriteString(tmp, line); err != nil {
+			log.Printf("Preparation: Error during write \"%s\" to tmp file, err: %s\n", line, err.Error())
+			return err
+		}
+		newLine = true
+	}
+	if scanner.Err() != nil {
+		return scanner.Err()
+	}
+
+	if err = os.Rename(tmp.Name(), filePath); err != nil {
+		log.Printf("Preparation: Error during rename temporary file, err: %s\n", err.Error())
+		return err
+	}
+	return nil
+}
+
+// createTempFile creates temporary file near with originalFile
+func createTempFile(originalFilePath string) (*os.File, error) {
+	// all folders which are included in filePath
+	filePathSlice := strings.Split(originalFilePath, pathSeparatorPattern)
+	fileName := filePathSlice[len(filePathSlice)-1]
+
+	tmpFileName := fmt.Sprintf("%s_%s", tmpFileSuffix, fileName)
+	tmpFilePath := strings.Replace(originalFilePath, fileName, tmpFileName, 1)
+	return os.Create(tmpFilePath)
+}
+
+// addNewLine add new line to tmp file
```

Comment:
```suggestion
// addNewLine adds a new line to tmp file
```

Reply:
Seems like this function adds a new line at the end of the file – can we mention it in the docstring?